### ISSUE TITLE
feat(crossasset): add GPU training via ztensor (parity with PatchTST)

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#312]
## Context

The `crossasset` package implements a cross-attention model for multi-source classification, but training uses pure Go float64 math (manual matVecMul, softmax, etc.). Meanwhile, `timeseries/patchtst_gpu_train.go` has full GPU training via ztensor with CUDA graph capture, achieving 64x fewer kernel launches.

## Problem

Training the CrossAsset model on 26K samples × 12 sources × 37 features × 50 epochs takes hours on CPU. The same architecture in PyTorch on GPU takes ~5 minutes.

## Request

Add GPU training to the CrossAsset model using the same ztensor + CUDA graph pattern as PatchTST:
- Forward pass through ztensor engine ops (matmul, softmax, layernorm, GELU)
- Backward pass with autograd or manual gradients
- CUDA graph capture for the training loop
- Batch workspace pre-allocation

## Current architecture

```
crossasset.Model
  - inputW [NSources][FeaturesPerSource * DModel] -- per-source projection
  - layers []layer -- cross-attention + FFN
  - headW [DModel * 3] -- classification head
  - Train() -- CPU SGD with manual gradients (head only)
  - Forward() -- CPU forward pass
  - Predict() -- CPU inference
```

The `Train()` method currently only backprops through the head layer. Full backprop through the cross-attention layers would also improve accuracy.

## Platform

- linux/arm64 (DGX Spark with Grace Hopper GPU)
- Zerfoo v1.38.5
- Go 1.26
